src/sentry/rules/conditions/first_seen_event.py

```diff
@@ -15,4 +15,7 @@ class FirstSeenEventCondition(EventCondition):
     label = 'An event is first seen'
 
     def passes(self, event, state):
-        return state.is_new
+        if self.rule.environment_id is None:
```

Comment:
Same comment above with regard to optional-but-not-really-optional `self.rule` here.